File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        int i=0;
        int j=0;
        int prod=1;
        int ans=0;

        while(j<nums.size())
        {
            prod *= nums[j];

           
                while(prod>=k && i<=j)
                {
                    prod=prod/nums[i];
                    i++;
                }
              
            ans=ans+(j-i+1);
            j++;
        }
        return ans;
    }
};

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int left =0 , right =0 ,ans =0;
        long prod =1 ;


        while(right < nums.length){

            prod *= nums[right];

            while(prod >= k && left <= right){
              prod = prod/ nums[left];
              left++;

            }
          
            ans = ans + (right-left+1);

            right++;
            
        }

        return ans;
    }
}